src/main/scala/no/ndla/draftapi/service/WriteService.scala

```diff
@@ -601,33 +601,63 @@ trait WriteService {
       }
     }
 
-    def partialPublish(id: Long, language: String, fallback: Boolean): Try[api.Article] =
-      partialPublish(id)._2.flatMap(article => converterService.toApiArticle(article, language, fallback))
+    def partialArticleFieldsUpdate(articleToPartialPublish: domain.Article,
+                                   articleFieldsToUpdate: Seq[PartialArticleFields.Value],
+                                   language: String): PartialPublishArticle = {
+      val newGrepCodes = if (articleFieldsToUpdate.contains(PartialArticleFields.grepCodes)) {
+        Some(articleToPartialPublish.grepCodes)
+      } else None
+      val newLicense = if (articleFieldsToUpdate.contains(PartialArticleFields.license)) {
+        articleToPartialPublish.copyright.flatMap(c => c.license)
+      } else None
+      val newMetaDesc = if (articleFieldsToUpdate.contains(PartialArticleFields.metaDescription)) {
+        if (language == "all") Some(articleToPartialPublish.metaDescription)
```

Comment:
Kunne sikkert vært smart å bruke `Language.AllLanguages` variabelen bare sånn for sikkerhetsskyld her, men er ikke så veldig farlig.